oap-server/server-tools/data-generator/src/main/java/org/apache/skywalking/restapi/SegmentGeneratorHandler.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.skywalking.restapi;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.function.IntConsumer;
+import java.util.stream.IntStream;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.util.EventLoopGroups;
+import com.linecorp.armeria.server.annotation.Default;
+import com.linecorp.armeria.server.annotation.Delete;
+import com.linecorp.armeria.server.annotation.Get;
+import com.linecorp.armeria.server.annotation.Param;
+import com.linecorp.armeria.server.annotation.Post;
+import com.linecorp.armeria.server.annotation.ProducesJson;
+import com.linecorp.armeria.server.annotation.RequestObject;
+import org.apache.skywalking.oap.server.core.CoreModule;
+import org.apache.skywalking.oap.server.core.source.Segment;
+import org.apache.skywalking.oap.server.core.source.SourceReceiver;
+import org.apache.skywalking.oap.server.library.module.ModuleManager;
+import io.netty.channel.EventLoopGroup;
+import io.netty.util.concurrent.Future;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class SegmentGeneratorHandler {
+    private final SourceReceiver sourceReceiver;
+    private final Map<String, Future<?>> futures = new ConcurrentHashMap<>();
+    private final EventLoopGroup eventLoopGroup = EventLoopGroups.newEventLoopGroup(10);
+
+    public SegmentGeneratorHandler(ModuleManager manager) {
+        sourceReceiver = manager.find(CoreModule.NAME).provider().getService(SourceReceiver.class);
+    }
+
+    @Post("/mock-data/segments/tasks")
+    public HttpResponse generateMockSegments(
+        @Default("0") @Param("size") int size,
+        @Default("0") @Param("qps") int qps,
```

Comment:
We need a time window param here. The generated segments should go into this window. 
If we pass the time window, the size could be 0.

Reply:
> We need a time window param here. The generated segments should go into this window.
> If we pass the time window, the size could be 0.

You should set the time window by specifying the time range (`min`, `max`) in the template. I'm not using the current timestamp as the segment/logs timestamp.

```
  "startTime": {
      "type": "randomInt",
      "min": "1647868381000",
      "max": "1647878402000"
  },
```

Reply:
I think documenting this should be done as it is important for time series telemetry data

Reply:
I noticed the type is `randomInt`. Does it mean the timestamps are random data points between min and max? Can you support generating monotonic time? The monotonically increasing timestamp is very important to banyandb's data model.

Reply:
monotonic as increasing should be more real. But ramdon should still be used for second(same day, hour, minute),because the timestamp of source services are never 100% sync, and segments finished unmonotonic slightly.